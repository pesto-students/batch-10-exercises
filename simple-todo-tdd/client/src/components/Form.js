import React from 'react'
import ReactDOM from 'react-dom'


const Form = ( props ) =>{

    const handleFormSubmit = (event) => {
        event.preventDefault();
        alert("You are submitting " + this.state.username);
      }

    return (
        <div>
        <h1> Insert New TODO </h1>
        <form onSubmit="handleFormSubmit">
            <lable for="NewTodo">TODO NAME </lable>
            <input type="text" id="NewTodo" name="eventName"/>
            <lable for="NewTodo">TODO Time </lable>
            <input type="text" id="NewTodo" name="time" />
            <button type="submit" > Save </button>
        </form>
        </div>
    )
}
 export default Form;