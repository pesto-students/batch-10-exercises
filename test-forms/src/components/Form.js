import React, { Component } from 'react';

import '../styles/Form.css';
// import api from '../api';

export default class Form extends Component {
  state = {
    name: '',
  }
  render() {
    return (
      <form data-testid="addUserForm" onSubmit={this.handleSubmit}>
        <h2>Request Information</h2>
        <input data-testid="name" type="text" placeholder="Name" value={this.state.name} />
        <div>
          <input data-testid="checked" type="checkbox" />
          <p data-testid="promotionsP" className="promotions">Receive Promotions</p>
        </div>
        <button type="submit" data-testid="submitButton">Submit</button>
      </form>
    );
  }
}
