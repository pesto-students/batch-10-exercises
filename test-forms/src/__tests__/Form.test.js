import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Form from '../components/Form';
import api from '../api';

const updateInput = (wrapper, instance, newValue) => {
  const input = wrapper.find(instance);
  input.simulate('change', {
    currentTarget: { value: newValue },
  });
  return wrapper.find(instance);
};

describe('<Form />', () => {
  test('receive promotions default is true', () => {
    const wrapper = shallow(<Form />);
    const promotionInput = wrapper.find('[data-testid="checked"]');
    expect(promotionInput.props().checked).toBe(true);
  });

  test('allows user to fill out form', () => {
    const wrapper = shallow(<Form />);
    const nameInput = updateInput(wrapper, '[data-testid="name"]', 'Pesto');
    const emailInput = updateInput(wrapper, '[data-testid="email"]', 'test@gmail.com');
    const numberInput = updateInput(wrapper, '[data-testid="number"]', '8018882321');
    wrapper.find('[data-testid="checked"]').simulate('click');

    expect(nameInput.props().value).toBe('Pesto');
    expect(emailInput.props().value).toBe('test@gmail.com');
    expect(numberInput.props().value).toBe('8018882321');
    expect(wrapper.find('[data-testid="checked"]').props().checked).toBe(false);
  });

  test('submits the form', () => {
    jest.spyOn(api, 'addUser').mockImplementation(() => Promise.resolve({ data: 'New User Added' }));
    const wrapper = shallow(<Form />);
    updateInput(wrapper, '[data-testid="name"]', 'Pesto');
    updateInput(wrapper, '[data-testid="email"]', 'test@gmail.com');
    updateInput(wrapper, '[data-testid="number"]', '8018882321');
    wrapper.find('[data-testid="addUserForm"]').simulate('submit', { preventDefault: () => {} });

    expect(api.addUser).toHaveBeenCalledWith('Pesto', 'test@gmail.com', '8018882321');
  });

  test('matches saved snapshot', () => {
    const wrapper = shallow(<Form />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
