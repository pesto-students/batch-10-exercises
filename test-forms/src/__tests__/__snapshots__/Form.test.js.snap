// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> matches saved snapshot 1`] = `
<form
  data-testid="addUserForm"
>
  <h2>
    Request Information
  </h2>
  <input
    data-testid="name"
    placeholder="Name"
    type="text"
    value=""
  />
  <div>
    <input
      data-testid="checked"
      type="checkbox"
    />
    <p
      className="promotions"
      data-testid="promotionsP"
    >
      Receive Promotions
    </p>
  </div>
  <button
    data-testid="submitButton"
    type="submit"
  >
    Submit
  </button>
</form>
`;
